src/libData/AccountData/AccountStore.cpp

```diff
@@ -85,11 +85,20 @@ AccountStore& AccountStore::GetInstance() {
   return accountstore;
 }
 
+bool AccountStore::Serialize(vector<unsigned char>& src,
+                             unsigned int offset) const {
+  LOG_MARKER();
+
+  shared_lock<shared_timed_mutex> lock(m_mutexPrimary);
```

Comment:
Why other places use unique_lock but here use shared_lock for m_mutexPrimary?

Reply:
Because this is the only function we allow multithread accessing concurrently.